eva/binder/binder_utils.py

```diff
@@ -134,3 +138,74 @@ def resolve_alias_table_value_expression(node: FunctionExpression):
     assert len(node.alias.col_names) == len(
         node.output_objs
     ), f"""Expected {len(node.output_objs)} output columns for {node.alias.alias_name}, got {len(node.alias.col_names)}."""
+
+
+def handle_bind_extract_object_function(
+    node: FunctionExpression, binder_context: StatementBinderContext
+):
+    """Handles the binding of extract_object function.
+        1. Bind the source video data
+        2. Create and bind the detector function expression using the provided name.
+        3. Create and bind the tracker function expression.
+            Its inputs are id, data, output of detector.
+        4. Bind the EXTRACT_OBJECT function expression and append the new children.
+        5. Handle the alias and populate the outputs of the EXTRACT_OBJECT function
+
+    Args:
+        node (FunctionExpression): The function expression representing the extract object operation.
+        binder_context (StatementBinderContext): The context object used to bind expressions in the statement.
+
+    Raises:
+        AssertionError: If the number of children in the `node` is not equal to 3.
+    """
+    assert (
+        len(node.children) == 3
+    ), f"Invalid arguments provided to {node}. Example correct usage, (data, Detector, Tracker)"
+
+    # 1. Bind the source video
+    video_data = node.children[0]
+    binder_context.bind(video_data)
+
+    # 2. Construct the detector
+    # convert detector to FunctionExpression before binding
+    # eg. YoloV5 -> YoloV5(data)
+    detector = FunctionExpression(None, node.children[1].col_name)
+    detector.append_child(video_data.copy())
+    binder_context.bind(detector)
+
+    # 3. Construct the tracker
+    # convert tracker to FunctionExpression before binding
+    # eg. ByteTracker -> ByteTracker(id, data, labels, bboxes, scores)
+    tracker = FunctionExpression(None, node.children[2].col_name)
+    # create the video id expression
+    columns = get_video_table_column_definitions()
+    tracker.append_child(
+        TupleValueExpression(
+            col_name=columns[1].name, table_alias=video_data.table_alias
+        )
+    )
+    tracker.append_child(video_data.copy())
+    binder_context.bind(tracker)
+    # append the bound output of detector
+    for obj in detector.output_objs:
```

Comment:
Would the object detection be run on every frame? norfair supports the period operator, but I'm curious how we can combine that with our operators (e.g., SAMPLE) to run detections only once every few frames and associate tracks.

Reply:
Nice observation! Let me think about it.

Reply:
Does the current implementation already work with SAMPLE? The frames passed to the object detector is after sampling.

Reply:
It will work, however incorrectly. Tracker won't know that we are skipping frames.

Reply:
I was sending `frame_id` to the `tracker` for such use cases. Fixed it!